src/connection.rs

```diff
@@ -29,13 +34,16 @@ pub enum ConnectionError {
 }
 
 impl Connection {
-    pub async fn new(id: usize, config: Arc<Config>) -> Result<Connection, ConnectionError> {
+    pub async fn new(id: usize, config: Arc<Config>, tx: Sender::<Histogram::<u64>>) -> Result<Connection, ConnectionError> {
```

Comment:
Instead of a new `Sender` in `new`, we can return `Histogram` from `start` and add it in main after a connection future is resolved.